### ISSUE TITLE
CPU goes 100% with Tmux

### ISSUE THREAD

**Issue by alexksikes** [fish-shell/fish-shell#1097]
Suppose fish is your default shell in Tmux. If you use Fish to create or attach to a Tmux  session, and then close the terminal without previously detaching, a fish process uses 100% of a CPU core. However, if you create or attach from bash, the issue does not arise.

I am using latest compiled version of Tmux and Fish running under updated version of Debian 7.

**Comment by krader1961** [comment]
This is pretty old and I've never seen it in using fish over 1.5 years with tmux so I'm closing it as unreproducible.

**Comment by jimbobmcgee** [comment]
I've just had this (or something similar), running from within `tmux`.  I think it is because my SSH client crashed and did not shut down properly, but I have an errant `fish` process running 100% CPU.

Not sure how to interpret an `strace`, but the following appears to be the repeating detail:

````
read(0, "", 1)                          = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
write(1, "fish: Process 25411, 'xargs' 'pg"..., 105) = -1 EIO (Input/output error)
fstat(1, {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 1), ...}) = 0
fstat(2, {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 1), ...}) = 0
fstat(1, {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 1), ...}) = 0
fstat(2, {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 1), ...}) = 0
select(6, [0 3 5], NULL, NULL, NULL)    = 1 (in [0])
rt_sigprocmask(SIG_BLOCK, [CHLD], [], 8) = 0
````

I got this while trying to debug a 100% CPU issue with either `vlock` or `tlock` used as the `lock-command` in `tmux`, if the SSH client bombs.  I've been testing by killing my SSH client uncleanly, which reliably repeats the issue with both `vlock` and `tlock` but, one time, I saw it with the `fish` process instead.  

When I reconnected, I could see a `fish` process with 100% CPU in `top`.  Running `ps -ef | grep fish` showed the `fish` process with no tty and parent PID of 1:

    root     25032     1 88 17:51 ?        00:33:58 /usr/bin/fish

I'm not certain its a fish issue (or at least not just a fish issue), as my original concern was with `tmux lock`ed sessions.

`fish` and `tmux` from EPEL and Centos 7 Base, respectively:

    Linux server.local 3.10.0-514.26.2.el7.x86_64 #1 SMP Tue Jul 4 15:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
    CentOS Linux release 7.3.1611 (Core)
    fish, version 2.3.1     (ver: 2.3.1 / rel: 1.el7 / repo: EPEL)
    tmux 1.8                (ver: 1.8 / rel: 4.el7 / repo: base)

`fish` is my login shell (via `chsh -s /usr/bin/fish`) and `tmux` is invoked from within an /etc/fish/conf.d script:

    if test -z "$TMUX"; and            \
       status --is-login; and          \
       status --is-interactive; and    \
       test -x (command -s tmux)
      if tmux has-session -t "$USER" 2>/dev/null
        # connect to the existing session
        exec tmux new-session -t "$USER"
      else
        # start a brand new session
        exec tmux new-session -s "$USER"
      end
    end

**Comment by krader1961** [comment]
@jimbobmcgee, That's definitely due to a bug in fish. I've fixed several relevant bugs since the fish 2.3.1 release. Can you reproduce using fish 2.6.0 or fish built from git Integration_2.7.0 branch?

**Comment by jimbobmcgee** [comment]
@krader1961 - to be honest, I haven't tried yet.  In fairness, I also haven't been able to recreate in 2.3.1, using my *killing the SSH client* method.  Works every time for `vlock` or `tlock`, but `fish` is usually OK.

For my sins, I'm trying to standardise my build around `yum update`ables (i.e. base, epel, etc.), as I won't be the eventual maintainer of these machines.  I appreciate this is my restriction, not yours, and figured the older version might be at least part of the problem.

I just saw your comment re not having seen it in 1.5 years and thought that you might like to know it's still possible to do a clean, 'brand new' build and experience these problems!

I guess my issue is with getting a newer version of fish into EPEL7.  I'm not entirely sure how to approach them on the matter, or whether they would even care.  I wasn't expecting EPEL to be bleeding-edge, anyway.
